.gitlab-ci.yml

```diff
@@ -71,16 +71,22 @@ build-gem:
   image: $DOCKER_REGISTRY/images/mirror/ruby:3.2.2
   stage: package
   script:
+    - mkdir -p pkg tmp
     - |
-      if [ -z "$CI_COMMIT_TAG" ]; then
+      if [ -n "$CI_COMMIT_TAG" ]; then
+        .gitlab/check_gem_presence.sh
+        VERSION=${CI_COMMIT_TAG#v}
+        gem fetch datadog --version $VERSION
```

Comment:
Nice, now it uses the exact same artifact in both places, with guaranteed no difference.

Reply:
This is because... run release process, which at the end creates the release/tag, which then triggers the gitlab job?

this seems definitely obvious what is happening, just asking to confirm I am correct.

Reply:
Yes. After the tag is pushed from the workflow, this tag push event starts a GitLab pipeline for building SSI artifacts. I use polling for the gem artifacts to make sure the single source of truth and cover the window of unavailability